src/core/middleware/security.js

```diff
@@ -40,14 +40,18 @@ export function csp({ _config = config, noScriptStyles, _log = log } = {}) {
         .createHash('sha256')
         .update(noScriptStyles)
         .digest('base64');
+
       const cspValue = `'sha256-${hash}'`;
-      if (
-        cspConfig.directives &&
-        !cspConfig.directives.styleSrc.includes(cspValue)
-      ) {
-        cspConfig.directives.styleSrc.push(cspValue);
+      const { directives } = cspConfig;
+
+      if (directives && !directives.styleSrc.includes(cspValue)) {
+        const styleSrc = [...directives.styleSrc];
+        styleSrc.push(cspValue);
+
+        config.util.extendDeep(cspConfig, { directives: { styleSrc } });
```

Comment:
Oof, I didn't know we were mutating. Do we really need to mutate the global config here? Is it referenced elsewhere? A quick grep says no. Mutating the global config could be confusing if you were looking at config files and wondering how the new values got there.

I'd suggest an approach where we copy the config and then mutate a copy:

```js
import deepcopy from 'deepcopy';
...
const cspConfig = deepcopy(_config.get('CSP');
...
cspConfig.directives.styleSrc.push(cspValue);
```

Reply:
Isn't it the purpose of this middleware to mutate the config?

Reply:
ah no, I think I now understand what you meant.